From 65e63d883a692338a3c28e086602521c70ba38ee Mon Sep 17 00:00:00 2001
From: hscham <hscham@chromium.org>
Date: Wed, 22 Dec 2021 15:19:25 +0900
Subject: [PATCH] base::TaskAnnotator::RunTask takes pending task as ref not
 ptr

Forward compatibility patch for the change in r935463
(crrev.com/c/3220334).

Change-Id: I2a66455e816c36249bb778f3b9031226d491430e
---
 base/task/common/task_annotator.cc | 5 +++++
 base/task/common/task_annotator.h  | 2 ++
 2 files changed, 7 insertions(+)

diff --git a/base/task/common/task_annotator.cc b/base/task/common/task_annotator.cc
index 63ec2abb1..b3a43895f 100644
--- a/base/task/common/task_annotator.cc
+++ b/base/task/common/task_annotator.cc
@@ -189,6 +189,11 @@ void TaskAnnotator::RunTask(const char* trace_event_name,
   debug::Alias(&task_backtrace);
 }
 
+void TaskAnnotator::RunTask(const char* trace_event_name,
+                            PendingTask& pending_task) {
+  RunTask(trace_event_name, &pending_task);
+}
+
 uint64_t TaskAnnotator::GetTaskTraceID(const PendingTask& task) const {
   return (static_cast<uint64_t>(task.sequence_num) << 32) |
          ((static_cast<uint64_t>(reinterpret_cast<intptr_t>(this)) << 32) >>
diff --git a/base/task/common/task_annotator.h b/base/task/common/task_annotator.h
index a4fa4bec6..c0e4c37f6 100644
--- a/base/task/common/task_annotator.h
+++ b/base/task/common/task_annotator.h
@@ -48,6 +48,8 @@ class BASE_EXPORT TaskAnnotator {
   // Run a previously queued task.
   void NOT_TAIL_CALLED RunTask(const char* trace_event_name,
                                PendingTask* pending_task);
+  void NOT_TAIL_CALLED RunTask(const char* trace_event_name,
+                               PendingTask& pending_task);
 
   // Creates a process-wide unique ID to represent this task in trace events.
   // This will be mangled with a Process ID hash to reduce the likelyhood of
-- 
2.34.1.307.g9b7440fafd-goog

